fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/FabricItem.java

```diff
@@ -119,15 +120,27 @@ default ItemStack getRecipeRemainder(ItemStack stack) {
 	 * <p>Note that this method is only called <em>after</em> the {@link EnchantmentEvents#ALLOW_ENCHANTING} event, and
 	 * only if none of the listeners to that event override the result.</p>
 	 *
-	 * @param stack the current stack
+	 * @param stack       the current stack
 	 * @param enchantment the enchantment to check
-	 * @param context the context in which the enchantment is being checked
+	 * @param context     the context in which the enchantment is being checked
 	 * @return whether the enchantment is allowed to apply to the stack
 	 */
 	default boolean canBeEnchantedWith(ItemStack stack, Enchantment enchantment, EnchantingContext context) {
 		return enchantment.isAcceptableItem(stack);
 	}
 
+	/**
+	 * Retrieves the normal 'lifespan' of this item when it is dropped on the ground
+	 * as a EntityItem. This is in ticks, standard result is 6000, or 5 mins.
+	 *
+	 * @param stack The current ItemStack
+	 * @param world The level the entity is in
+	 * @return The normal lifespan in ticks.
+	 */
+	default int getEntityLifeSpan(ItemStack stack, World world) {
+		return 6000;
```

Comment:
Can we please use `ItemEntity.DESPAWN_AGE` and get rid of the weird constants? This is a private field, so we will need (T)AW or MixinAccessor, but this is minor

Reply:
I suggest using TAW because in my opinion it is useful to open this field to everyone

Reply:
I would disagee, as anyone wanting to use this value should now be getting it via the itemstack.

Reply:
> I would disagee, as anyone wanting to use this value should now be getting it via the itemstack.

Sorry if I sound stupid I am not that good with mixins, how would I access the const if it's in a mixin accessor?

Reply:
I would use a mixin accessor.

Reply:
@modmuss50 I would assume like this:
![image](https://github.com/FabricMC/fabric/assets/84230130/1d0df2b4-c6de-44d8-a993-89c40dc9c178)

Reply:
Yes 👍

Reply:
ok tests coming tmrw.